### ISSUE TITLE
CRC-64 does not produce valid CRC-64-ECMA hash

### ISSUE THREAD

**Issue by KuznetsovRoman** [ClickHouse/ClickHouse#68981]
**Company or project name**

Yandex

**Describe what's wrong**

[CRC64](https://clickhouse.com/docs/en/sql-reference/functions/string-functions#crc64) function returns not valid "CRC-64-ECMA" hash.

"CRC-64-ECMA" of "1" should be "3039664240384658157"

A colleague found out it is because ClickHouse is lacking two inversions on input and output values of CRC function. In [his example on Godbolt](https://godbolt.org/z/v69v5rGd4) he marked those two lines with exclamation marks where ClickHouse should have inversion, but it doesn't. If those two inversions are skipped, like in ClickHouse, result is "3855837706121835956", which is wrong.

ClickHouse MRE: https://fiddle.clickhouse.com/2dfe57b2-c5ca-49a0-af65-79469d74d29b

**Does it reproduce on the most recent release?**

Yes

**How to reproduce**

https://fiddle.clickhouse.com/2dfe57b2-c5ca-49a0-af65-79469d74d29b

* Which ClickHouse server version to use
Latest
* Queries to run that lead to an unexpected result
`SELECT crc64('1');`

**Expected behavior**

I expect to get "3039664240384658157" and not "3855837706121835956"

**Additional context**

Since fixing this would be a major change, i suggest to add a new crc64 function and describe the difference between the new (with two inversions) and existing one (without these two inversions)

**Comment by pet74alex** [comment]
[Here is](https://go.dev/src/hash/crc64/crc64.go) an implementation of the same functionality in GoLang, which I (the colleague mentioned above) compared yesterday with ClickHouse CRC code, lines 138 and 178 contain the crc value inversions those are absent in the ClickHouse code.  Also there´s an additional speed optimization at lines 140-167 (manual loop unrolling) which could possibly be applicable for ClickHouse too.

**Comment by nikitamikhaylov** [comment]
@azat WDYT? Seems like it was added initially by you here: https://github.com/ClickHouse/ClickHouse/pull/7480

I've taken a look at a kernel code and seems like the implementation you are referencing to (and which is implemented in Golang) is in fact the CRC64 Rocksoft: https://github.com/torvalds/linux/blob/86987d84b968b69a610fd00ab9006c13db193b4e/lib/crc64.c#L73

**Comment by KuznetsovRoman** [comment]
@nikitamikhaylov based on https://github.com/gburca/RocksoftCRC/blob/master/Rocksoft/CRC_Explained.txt#L2004, it looks like "Rocksoft^TM" is not "another specification for CRC64", its just parameterized version of the crc64 algorithm

**Comment by nikitamikhaylov** [comment]
Yes, but in the kernel's source code it does have an inversion https://github.com/torvalds/linux/blob/86987d84b968b69a610fd00ab9006c13db193b4e/lib/crc64.c#L78

My assumption is more like a wild guess for now, just trying to figure our different implementations.
